File: golang/1689-partitioning-into-minimum-number-of-deci-binary-numbers.go
package golang

func minPartitions(n string) int {
    digits := len(n)
    result := 0
    for i := 0; i < digits; i++ {
        d := int(n[i] - '0')
        if d > result {
            result = d
        }
    }
    return result
}

File: php/1689-partitioning-into-minimum-number-of-deci-binary-numbers.php
class Solution {

    /**
    * @param String $n
    * @return Integer
    */
    function minPartitions($n) {
        $result = 0;
        for ($i = 0; $i < strlen($n); $i++) {
            $result = max($result, $n[$i]);
        }
        return $result;
    }
}